sycl/include/sycl/ext/oneapi/bindless_images.hpp

```diff
@@ -910,6 +910,22 @@ DataT fetch_image(const sampled_image_handle &imageHandle [[maybe_unused]],
 #endif
 }
 
+template <typename DataT>
+std::enable_if_t<std::is_same_v<DataT, float4> || std::is_same_v<DataT, int4> ||
+                     std::is_same_v<DataT, uint4>,
+                 DataT>
+gather_image(const sampled_image_handle &imageHandle [[maybe_unused]],
+             const float2 &coords [[maybe_unused]],
+             const unsigned i [[maybe_unused]]) {
+#if defined(__SYCL_DEVICE_ONLY__) && defined(__NVPTX__)
+  return __invoke__SampledImageGather<DataT>(
+      CONVERT_HANDLE_TO_SAMPLED_IMAGE(imageHandle.raw_handle, float2::size()),
+      coords, i);
+#else
+  assert(false); // Gather_image is only currently supported on the cuda backend
```

Comment:
I'm not sure all device targets would be happy seeing this. Could we return some default value on targets that don't support it? If we can ensure that it only runs on these (see previous comment) the default value should never be returned.

Reply:
Yeah I think the ideal situation is that an aspect is attached to the api in the way that you suggest.
I think this actually applies more generally to a lot of existing APIs, that may are calling unimplemented spirv functions on a lot of the backends. @intel/bindless-images-reviewers do you have any thoughts on this?